plugins/bundle/plugin.go

```diff
@@ -62,7 +62,7 @@ type Plugin struct {
 	downloaders       map[string]Loader
 	logger            logging.Logger
 	mtx               sync.Mutex
-	cfgMtx            sync.Mutex
+	cfgMtx            sync.RWMutex
```

Comment:
We need to hold the write lock while updating the config. Also if you could look at the test failures that would be great. Thanks.

Reply:
Thanks. I have addressed it. 
Could you please have a look?